sdks/python/apache_beam/io/range_trackers.py

```diff
@@ -31,15 +31,17 @@
 class OffsetRange(object):
 
   def __init__(self, start, stop):
-    if stop < start:
+    if start >= stop:
       raise ValueError(
-          'Stop offset must not be smaller than the start offset. '
+          'Start offset must be smaller than the stop offset. '
           'Received %d and %d respectively.', start, stop)
     self.start = start
     self.stop = stop
 
   def __eq__(self, other):
-    assert isinstance(other, OffsetRange)
+    if not isinstance(other, OffsetRange):
+      return False
+
     return self.start == other.start and self.stop == other.stop
 
   def __ne__(self, other):
```

Comment:
Do you want to change the assert in `__ne__` to return True, similar to `__eq__`?